catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -246,10 +247,10 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SourceResponse registryQueryResponse = super.query(request);
         for (Result singleResult : registryQueryResponse.getResults()) {
-            if (singleResult.getMetacard()
-                    .getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                    .getValue()
-                    .toString()
+            if (!RegistryUtility.isRegistryMetacard(singleResult.getMetacard())) {
+                break;
+            }
+            if (RegistryUtility.getRegistryId(singleResult.getMetacard())
```

Comment:
We discussed -> I was convinced, changed. (Overall removes number of evaluations by 1)